### PR TITLE
Convert redisson Groovy tests to Java

### PR DESCRIPTION
**Pull request by pg-yang**
related https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7195

### CODE REVIEW

File: instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("SET")
+                        .hasKind(CLIENT)
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1")
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_NAME, "localhost")
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_PORT, (long) port)
+                        .hasAttribute(SemanticAttributes.DB_SYSTEM, "redis")
+                        .hasAttribute(SemanticAttributes.DB_STATEMENT, "SET foo ?")
+                        .hasAttribute(SemanticAttributes.DB_OPERATION, "SET")));
```
Comment (mateuszrzeszutek) [inline instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java:119]: You can use `hasAttributesSatisfyingExactly()` to assert the exact set of attributes observed:
```suggestion
                        .hasAttributesSatisfyingExactly(
                          equalTo(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1"),
                          equalTo(SemanticAttributes.NET_SOCK_PEER_NAME, "localhost"),
                          equalTo(SemanticAttributes.NET_SOCK_PEER_PORT, (long) port),
                          equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
                          equalTo(SemanticAttributes.DB_STATEMENT, "SET foo ?"),
                          equalTo(SemanticAttributes.DB_OPERATION, "SET")));
```

File: instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  @Test
+  void scheduleCallable()
+      throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    RScheduledExecutorService executorService = redisson.getExecutorService("EXECUTOR");
+    Object o = invokeSchedule(executorService);
+    Object getTaskId = o.getClass().getMethod("getTaskId").invoke(o);
+    assertThat(getTaskId).isNotNull();
+  }
+
+  private static Object invokeSchedule(RScheduledExecutorService executorService)
```
Comment (mateuszrzeszutek) [inline instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java:167]: Just wondering: does this method change in different versions of redisson? Why use reflection here?

File: instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonClientTest.java [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+  @Test
+  void batchCommand()
+      throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+    RBatch batch = createBatch(redisson);
+    assertThat(batch).isNotNull();
+    batch.getBucket("batch1").setAsync("v1");
+    batch.getBucket("batch2").setAsync("v2");
+    // Adapt different method signature:
+    // `BatchResult<?> execute()` and `List<?> execute()`
+    invokeExecute(batch);
```
Comment (mateuszrzeszutek) [inline instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonClientTest.java:139]: 👍

File: instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java (hunk 1 above)
Comment (pg-yang) [reply instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java]: Done

File: instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java (hunk 2 above)
Comment (pg-yang) [reply instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java:176]: I have added a comment to explain it.  But I'm not sure the grammar is correct.

Comment (pg-yang) [reply instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java:181]: By the way, should I rebase the branch to combine commits into one?  I think the merge work could do it as well

**Review by mateuszrzeszutek — APPROVED** [review]
Thanks @pg-yang !

**Pull request merged**